### PR TITLE
Report closed shards and unavailable ingesters when requesting open shards

### PR DESCRIPTION
**Pull request by guilload**

### CODE REVIEW

**Comment by guilload** [comment]
Ready for review. With this change, routers now report closed shards and unavailable ingesters to the control plane. The following PR will implement the logic that fences or closes reported unavailable or closed shards.

File: quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs [hunk 1]
```diff
@@ -591,17 +592,17 @@ mod tests {
...
                 index_id: "test-index-1".to_string(),
                 source_id: source_id.to_string(),
-                closed_shards: Vec::new(),
             },
         ];
+        let closed_shards = Vec::new();
         let unavailable_ingesters = vec!["test-ingester-0".to_string()];
         let request = GetOrCreateOpenShardsRequest {
             subrequests,
+            closed_shards,
```
Comment (etolbakov) [inline quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs:605]: just out of curiosity, why this is not inlined? convention?

Comment (fulmicoton) [reply quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs:605]: No reasons but it does not matter.

File: quickwit/quickwit-ingest/src/ingest_v2/ingest.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Replication
+
+### Settings
+
+- ingest request timeout (30s), `Itimeout`
+- persist request timeout (6s), `Ptimeout`
+- replicate request timeout (3s), `Rtimeout`
+- number of persist attempts (5), `k`
```
Comment (fulmicoton) [inline quickwit/quickwit-ingest/src/ingest_v2/ingest.md:8]: super helpful

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**